test_scripts/Smoke/API/012_PerfomInteraction_PositiveCase.lua

```diff
@@ -0,0 +1,428 @@
+	config.application1.registerAppInterfaceParams.appID .. "_" .. config.deviceMAC .. "/"
+local ImageValue = {
+			value = storagePath .. "icon.png",
+			imageType = "DYNAMIC",
+    }
+local function PromptValue(text)
+	local tmp = {
+		{
+			text = text,
+			type = "TEXT"
+		}
+	}
+	return tmp
+end
+local initialPromptValue = PromptValue(" Make  your choice ")
+local helpPromptValue = PromptValue(" Help   Prompt  ")
+local timeoutPromptValue = PromptValue(" Time  out  ")
+local vrHelpvalue = {
+		{
+			text = "  New  VRHelp  ",
+			position = 1,
+			image = ImageValue
+		}
+	}
+local requestParams = {
+	initialText = "StartPerformInteraction",
+	initialPrompt = initialPromptValue,
+	interactionMode = "BOTH",
+	interactionChoiceSetIDList = {
+		100, 200, 300
+	},
+	helpPrompt = helpPromptValue,
+	timeoutPrompt = timeoutPromptValue,
+	timeout = 5000,
+	vrHelp = vrHelpvalue,
+	interactionLayout = "ICON_ONLY"
+}
+
+--! @setChoiseSet: Creates Choice structure
+--! @parameters:
+--! choiceIDValue - Id for created choice
+--! @return: table of created choice structure
+local function setChoiseSet(choiceIDValue)
+	local temp = {
+		{
+			choiceID = choiceIDValue,
+			menuName ="Choice" .. tostring(choiceIDValue),
+			vrCommands = {
+				"VrChoice" .. tostring(choiceIDValue),
+			},
+			image = {
+				value ="icon.png",
+				imageType ="STATIC",
+			}
+		}
+	}
+	return temp
+end
+
+--! @SendOnSystemContext: OnSystemContext notification
+--! @parameters:
+--! self - test object,
+--! ctx - systemContext value
+--! @return: none
+local function SendOnSystemContext(self, ctx)
+	self.hmiConnection:SendNotification("UI.OnSystemContext",
+		{ appID = self.applications["Test Application"], systemContext = ctx })
+end
+
+--! @setExChoiseSet: ChoiceSet structure for UI.PerformInteraction request
+--! @parameters:
+--! choiceIDValues - value of choice id
+--! @return: none
+local function setExChoiseSet(choiceIDValues)
+	local exChoiceSet = { }
+	for i = 1, #choiceIDValues do
+		exChoiceSet[i] =  {
+			choiceID = choiceIDValues[i],
+			image = {
+				value = "icon.png",
+				imageType = "STATIC",
+			},
+			menuName = "Choice" .. choiceIDValues[i]
+		}
+	end
+	return exChoiceSet
+end
+
+--! @ExpectOnHMIStatusWithAudioStateChanged_PI: Expectations of OnHMIStatus notification depending on the application
+--! type, HMI level and interaction mode
+--! @parameters:
+--! self - test object,
+--! request - interaction mode,
+--! timeout - timeout value for expectation OnHMIStatus,
+--! level - HMI level value,
+--! @return: none
+local function ExpectOnHMIStatusWithAudioStateChanged_PI(self, request, timeout, level)
+	if request == nil then  request = "BOTH" end
+	if level == nil then  level = "FULL" end
+	if timeout == nil then timeout = 10000 end
+	commonSmoke.AppType(config.application1.registerAppInterfaceParams.appHMIType)
+	if level == "FULL" then
+			if
+				self.isMediaApplication == true or
+				commonSmoke.HMITypeStatus.Navigation == true then
```

Comment:
@GetmanetsIrina In common module 'HMITypeStatus' table has 2 keys: NAVIGATION and COMMUNICATION.
So exact (uppercase) key values have to be used when using this table.

Reply:
@dboltovskyi, updated.